siatest/consensus/consensus_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/NebulousLabs/Sia/node"
+	"github.com/NebulousLabs/Sia/siatest"
+)
+
+// TestApiHeight checks if the consensus api endpoint works
+func TestApiHeight(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	testdir, err := siatest.TestDir(siatest.SiaTestingDir, t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a new server
+	testNode, err := siatest.NewNode(node.AllModules(testdir))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Send GET request
+	cg, err := testNode.GetConsensus()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check height
+	height := cg.Height
+
+	// Mine a block
+	if err := testNode.MineBlock(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Request height again
+	cg, err = testNode.GetConsensus()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check if height increased
+	if cg.Height != height+1 {
+		t.Fatal("Height should have increased by 1 block")
+	}
+
+	// Close the server and check error
+	defer func() {
+		if err := testNode.Close(); err != nil {
+			t.Fatal(err)
+		}
+	}()
```

Comment:
The code here could be compacted a little more. For example, pulling the `height` out of the `cg` should probably happen in the same codeblock, since it follows the same thought of extracting the height. And then when you check the height, getting the height a second time and performing the check can probably also be the same codeblock.

The `defer close` block should happen right after the call to `NewNode`, because we want to make sure we free resources at the same exact time that we allocate them (to minimize mistakes).